### PR TITLE
Add option to ignore GeoParquet, disable spatial autoloading when reading GeoParquet

### LINKED ISSUE #124: Auto Install/Load extensions is causing schema inference discrepancies
We use Duckdb for inferring Parquet file schemas. The latest update (1.1.1) has changed the default behavior of parquet schema inference if `spatial` extension is installed. The problem is that the nodejs module automatically installs/loads the extensions. Is there a way to disable/override this behavior? We do not want `spatial` extension to be installed. I see that the DUCKDB_EXTENSION_AUTOINSTALL_DEFAULT flag is set by default. Is there a way to unset it (maybe using an environment variable) ?

### LINKED ISSUE #423: Geoparquet column 'geometry' does not have geometry types
After upgrading to 1.1.0, it is no longer possible to open old parquet files created with WKB geometry.

`Invalid Input Error: Geoparquet column 'geometry' does not have geometry types`

The error occurs even when we do not select geometry columns. I assume that DuckDB is trying to automatically convert geometry columns to GEOMETRY type. Is there any way to disable this? Or maybe you can recommend some other way to handle this?

### PR DESCRIPTION
**Pull request by Maxxen**
Closes https://github.com/duckdb/duckdb-node/issues/124
Closes https://github.com/duckdb/duckdb_spatial/issues/423

This PR changes the GeoParquet reading behavior so that the spatial extension is no longer _auto-loaded_ when GeoParquet metadata is detected. Spatial was not marked as autoloadable and the way the load was issued did not respect auto loading settings either. If spatial would become auto-loadable in the future we can change this behavior back (but this time also respect auto loading settings).

This PR also adds a new setting to the parquet extension `enable_geoparquet_conversion = true (default)` which controls whether geometry data will be converted when reading/writing GeoParquet files (and the spatial extension is loaded). This enables you to "fall back" to reading writing parquet files normally even if they have geoparquet metadata. This is primarily useful when trying to read a geoparquet file that is incompatible with the current geoparquet support in duckdb, either because it uses an unsupported version/encoding or because it is non-standards compliant (or the metadata is broken in some other way).

Additionally, I've also moved over the geoparquet tests and test files into this repository for now. Since they `require spatial` they won't be invoked during our normal CI runs (I think?), but it still useful when testing locally.

### CODE REVIEW

File: extension/parquet/geo_parquet.cpp [hunk 1]
```diff
@@ -187,10 +194,8 @@ GeoParquetFileMetadata::TryRead(const duckdb_parquet::format::FileMetaData &file
 			}
 
 			// Check if the spatial extension is loaded, or try to autoload it.
-			const auto is_loaded = ExtensionHelper::TryAutoLoadExtension(context, "spatial");
-			if (!is_loaded) {
-				// Spatial extension is not available, we can't make use of the metadata anyway.
-				yyjson_doc_free(geo_metadata);
```
[the remark this answers is not in the document]
Comment (Maxxen) [reply extension/parquet/geo_parquet.cpp:193]: Good catch, thanks!

File: test/geoparquet/disabled.test [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# name: test/geoparquet/disabled.test
+# group: [geoparquet]
+
+require spatial
+
+require parquet
```
Comment (Mytherin) [inline test/geoparquet/disabled.test:6]: Can we add tests when spatial is not loaded as well?

Comment (Maxxen) [reply test/geoparquet/disabled.test:6]: Done!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**